src/js/relayclient/RelayClient.js

```diff
@@ -299,6 +299,15 @@ class RelayClient {
     const gasSponsor = options.gasSponsor || options.to
     const relayHub = await this.createRelayHubFromSponsor(gasSponsor)
 
+    // TODO: refactor! wrong instance is created for accidentally same method!
+    if (!utils.isSameAddress(gasSponsor, options.to)) {
+      const recipientHub = await this.createGasSponsor(options.to).methods.getHubAddr().call()
+
+      if (!utils.isSameAddress(relayHub._address, recipientHub)) {
+        throw Error('Sponsor and recipient RelayHub addresses do not match')
```

Comment:
The error message should be "wrong sponsor: uses different RelayHub than contact'

Reply:
Because you prefer your error messages obscure and incorrect?

Reply:
The error messages are not for us, but for the end-users - should explain the error, and preferably, when went wrong. The problem is that probably the user doesn't know when sponsor (or even "gasPayer") is - he might know the specific implementation.
he might also not know what we call "the contract" - but we don't have a way to give them better names...

Reply:
Still, if you don't know what is going on no amount of error messages will teach you everything, and if you have even a vogue idea this error should be clear enough.